plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb

```diff
@@ -47,6 +51,41 @@ def template
       end
     end
 
+    # process the annotations, and expand $ENV and $DEPLOY_GROUP for any annotation
+    # that begins with 'secret/'
+    def expand_secret_annotations
+      annotations.to_h.each do |annotation_name, secret_key|
+        if annotation_name.to_s.start_with?('secret/')
+          secret_key.gsub!(/\${ENV}/, @doc.deploy_group.environment.permalink)
+          secret_key.gsub!(/\${DEPLOY_GROUP}/, @doc.deploy_group.permalink)
+        end
+      end
+    end
+
+    # verify that each secret really exists and inform the user
+    # if it does not as the deployment will fail
+    def verify_secret_annotations
+      errors = []
+      annotations.to_h.each do |annotation_name, secret_key|
+        next unless annotation_name.to_s.start_with?('secret/')
+        begin
+          SecretStorage.read(secret_key)
+        rescue
```

Comment:
ok, got the last batch.

On Mon, Jun 6, 2016 at 2:48 PM, Michael Grosser <notifications@github.com>
wrote:

> In plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb
> <https://github.com/zendesk/samson/pull/1048#discussion_r65975780>:
>
> > +        if annotation_name.to_s.start_with?('secret/')
> > +          secret_key.gsub!(/\${ENV}/, @doc.deploy_group.environment.permalink)
> > +          secret_key.gsub!(/\${DEPLOY_GROUP}/, @doc.deploy_group.permalink)
> > +        end
> > +      end
> > +    end
> > +
> > +    # verify that each secret really exists and inform the user
> > +    # if it does not as the deployment will fail
> > +    def verify_secret_annotations
> > +      errors = []
> > +      annotations.to_h.each do |annotation_name, secret_key|
> > +        next unless annotation_name.to_s.start_with?('secret/')
> > +        begin
> > +          SecretStorage.read(secret_key)
> > +        rescue
>
> rescue a specific error instead of everything ?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/zendesk/samson/pull/1048/files/c8215817a7f0a3e32e6975017503ae320b3c0814#r65975780>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAJfonqqJMy4cPoZ7UFQLpZvQW5OFAwoks5qJJWugaJpZM4It_rG>
> .
>